recipes/7bitconf/all/conanfile.py

```diff
@@ -0,0 +1,144 @@
+from conan.tools.scm import Version
+import os
+
+required_conan_version = ">=1.53.0"
+
+
+class SevenBitConfConan(ConanFile):
+    name = "7bitconf"
+    homepage = "https://github.com/7bitCoder/7bitConf"
+    description = "7bitConf is a simple C++17 centralized configuration provider library."
+    topics = ("cpp17", "configuration", "provider", "configuration-files")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "header_only": [True, False],
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "header_only": False,
+        "shared": False,
+        "fPIC": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+    
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "14",
+            "msvc": "192",
+            "gcc": "8",
+            "clang": "7",
+            "apple-clang": "10",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
+
+    def configure(self):
+        if self.options.get_safe("shared") or self.options.header_only:
+            self.options.rm_safe("fPIC")
+        if self.options.header_only:
+            self.options.rm_safe("shared")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("taocpp-json/1.0.0-beta.13", transitive_headers=True, transitive_libs=True)
+
+    def package_id(self):
+        if self.info.options.header_only:
+            self.info.clear()
+
+    def validate(self):
+        compiler = self.settings.compiler
+        compiler_name = str(compiler)
+
+        if compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+
+        minimum_version = self._minimum_compilers_version.get(compiler_name, False)
+        if minimum_version and Version(compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"Requires compiler {compiler_name} minimum version: {minimum_version} with C++17 support."
+            )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        if not self.options.header_only:
+            tc = CMakeToolchain(self)
+            tc.variables["_7BIT_CONF_BUILD_EXAMPLES"] = False
+            tc.variables["_7BIT_CONF_BUILD_TESTS"] = False
+            tc.variables["_7BIT_CONF_BUILD_SINGLE_HEADER"] = False
+            tc.variables["_7BIT_CONF_INSTALL"] = True
+            tc.variables["_7BIT_CONF_LIBRARY_TYPE"] = self.getSevenBitConfLibraryType()
+            tc.generate()
+        cmake_deps = CMakeDeps(self)
+        cmake_deps.generate()
+
+    def getSevenBitConfLibraryType(self):
```

Comment:
This shouldn't be necessary. Conan is able to dynamically compute the ``package_type`` based on the options, so you can check directly either the options or the ``package_type``.

Reply:
Thank you, I didn't know about it, I have simplified this code with package_type check.

Reply:
There were some some problems with checking  package_type so I have reverted this code to previous version.

Reply:
@memsharded the package type here is not the recipe attribute, but a custom project's cmake definition. Indeed it needs to declare the package type.

Reply:
Understood, thanks for clarifying!